pyproject.toml

```diff
@@ -139,7 +138,7 @@ test-command = "bash {project}/tools/wheels/cibw_test_command.sh {project}"
 [tool.cibuildwheel.linux]
 manylinux-x86_64-image = "manylinux2014"
 manylinux-aarch64-image = "manylinux2014"
-before-build = "bash {project}/tools/wheels/cibw_before_build_linux.sh {project}"
+before-build = "bash {project}/tools/wheels/cibw_before_build.sh {project}"
 
 [tool.cibuildwheel.macos]
 before-build = "bash {project}/tools/wheels/cibw_before_build_macos.sh {project}"
```

Comment:
This line will override any config specified in `[tool.cibuildwheel]` (i.e. the new file you add won't get called). Ditto the specific windows file.

For me it makes sense to separate configuration into separate `cibw_before_build_platform` files, a single file can get convoluted.

Reply:
Perhaps there could be a common file, and individual files in this line, i.e.
```
before-build = "bash {project}/tools/wheels/cibw_before_build_common.sh; bash {project}/tools/wheels/cibw_before_build_macos.sh {project}"
```

Reply:
> This line will override any config specified in `[tool.cibuildwheel]` (i.e. the new file you add won't get called).

At this time, I only want to override the linux wheel building to prove the wheels work. If it works, we can think about how to generalize it in another PR/other commits.

Reply:
Ok. The changes you make here are also overridden in [wheels.yml](https://github.com/scipy/scipy/blob/85736e165b309ae57181c9c24f65c6a70b1eb7fa/.github/workflows/wheels.yml#L179)

Reply:
Oy. I will try removing the redundant line from `wheels.yml`.

Reply:
Having a duplicate is sometimes useful. e.g. if you want to develop locally you want things specified in pyproject.toml. If you want a fixed config for scipy/scipy to build wheels for public consumption you may want to do that in wheels.yml.

Reply:
> For me it makes sense to separate configuration into separate ...

OK, adopted in the latest commit

Reply:
> Having a duplicate is sometimes useful.

Hmm. On the other hand, it is a bit obscure and makes replicating the wheel build locally more difficult...

Reply:
Maybe we could document the ability to override the pyproject.toml values with local environment variables, and local specialized builds would do that

Reply:
Sure. The main reason to do so is e.g. employ specific package versions, etc.

Reply:
I like @mattip's idea. The duplication makes the code harder to understand, and I'd really like it to be as clean as possible. So comments only to "how do I tweak this if needed" are nicer than more bash code.